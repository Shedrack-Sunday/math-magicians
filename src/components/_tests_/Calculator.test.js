import React from 'react';
import renderer from 'react-test-renderer';
import Calculator from '../Calculator';
import '@testing-library/jest-dom/extend-expect';

describe('Test render Calculator component', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<Calculator />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
