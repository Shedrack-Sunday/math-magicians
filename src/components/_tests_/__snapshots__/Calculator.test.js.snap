// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test render Calculator component renders correctly 1`] = `
<div
  style={
    Object {
      "alignItems": "flex-start",
      "display": "flex",
      "flexDirection": "row",
      "justifyContent": "space-around",
      "margin": "0 auto",
      "width": "56rem",
    }
  }
>
  <h1
    style={
      Object {
        "color": "darkGray",
        "margin": "0",
      }
    }
  >
    Lets do some Math
  </h1>
  <div
    className="container"
  >
    <div
      className="result"
    />
    <button
      className="sc-bcXHqe hMpuZe btn-zero"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="sc-bcXHqe bNmRwW btn-equal"
      onClick={[Function]}
      type="button"
    >
      =
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-clear"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-toogle-sign"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-percentage"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="sc-bcXHqe bNmRwW btn-divider"
      onClick={[Function]}
      type="button"
    >
      /
    </button>
    <button
      className="sc-bcXHqe bNmRwW btn-multiply"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="sc-bcXHqe bNmRwW btn-substract"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="sc-bcXHqe bNmRwW btn-plus"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-one"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-two"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-three"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-four"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-five"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-six"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-seven"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-eight"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-nine"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="sc-bcXHqe hMpuZe btn-dot"
      onClick={[Function]}
      type="button"
    >
      .
    </button>
  </div>
</div>
`;
